client/cmdlfem4x.c

```diff
@@ -1213,19 +1213,27 @@ static int CmdEM4x05Dump(const char *Cmd) {
         usePwd = true;
 
     int success = PM3_SUCCESS;
+    int status;
     uint32_t word = 0;
-    PrintAndLogEx(NORMAL, "Addr | data   | ascii");
-    PrintAndLogEx(NORMAL, "-----+--------+------");
+    PrintAndLogEx(NORMAL, "Addr | data     | info");
+    PrintAndLogEx(NORMAL, "-----+----------+-------");
     for (; addr < 16; addr++) {
 
         if (addr == 2) {
             if (usePwd) {
-                PrintAndLogEx(NORMAL, " %02u | %08X", addr, pwd, word);
+                PrintAndLogEx(NORMAL, "  %02u | %08X |", addr, pwd, word);
```

Comment:
You are missing to print all vars.

Reply:
Yes, I copied the print from the read (single block function).  It was not printing the ascii, rather things like Lock.  What is the function to print the ascii, I can put back in.
I assume the write to file needs to be developed ?  or did that drop off somewhere as well ?

Reply:
output  headers should follow other LF outputs.    Look inside client.util.c  for hex / ascii prints.

blk |  data | ascii  | lck
--

The dump save?   Make an array of all found blocks. 
send them into   savefile,  savefileEML,    for JSON we would need to add some more description.  We could add the lock data in it.

Reply:
Yep, leave it with me, I will fix the headings etc.  and add the save.